lxd/api_internal_recover.go

```diff
@@ -177,6 +177,12 @@ func internalRecoverScan(s *state.State, userPools []api.StoragePoolsPost, valid
 					return response.SmartError(fmt.Errorf("Failed to initialise unknown pool %q: %w", p.Name, err))
 				}
 
+				// Populate configuration with default values
+				err := pool.Driver().FillConfig()
```

Comment:
So the commit ordering is a little odd here, as we are calling `FillConfig` before its been added to the storage interface and drivers. Please can you order the commits logically so they dont use resources before they exist.

Also, just a small nit, but can you end comments with a full stop. Thanks